#159987715 Tag articles on creation

emekafredy opened the pull request:
# What does this PR do?
Create endpoints for users profile

# Description of Task to be completed?
### Create tag model and migration file
### Write test for feature
### Create endpoint
- Create tags ``` POST api/articles/tags ```
### Modify endpoint
- Article creation with tags ``` POST api/articles ```

# How should this be manually tested?
- Clone the repo
- cd into the 'Thor-ah'
- Run 'npm install'
- Use the .env.sample format to create a .env file
- Run 'npm run migrate'
- Run 'npm run start:dev'

# Any background context you want to provide?
N/A

# What are the relevant pivotal tracker stories?
[159987715](https://www.pivotaltracker.com/story/show/159987715)

# Screenshots (if appropriate)
N/A

# Questions:
N/A

emekafredy committed:
first branch commit

emekafredy committed:
feat(articleTags): define endpoint for tag creation

- define and create a tag model with association to article
- write test for feature
- define endpoint for tag creation
- modify article creation function to add relevant tags to article (not more than 5)

[Delivers #159987715]

emekafredy committed:
bug(hound): fix hound issues

CEOehis commented on server/tests/controllers/articles.js:
```diff
@@ -179,7 +258,7 @@ describe('Articles controller', () => {
     });
 
-    it('should return error if slug not found', (done) => {
+    it('shouuld return error if slug not found', (done) => {
```
@emekafredy typo here and on line `237`

emekafredy replied:
Okay. Thanks. I will fix this

CEOehis commented on server/tests/controllers/articles.js:
```diff
@@ -63,18 +108,52 @@ describe('Articles controller', () => {
+          newArticleAlert.createdArticle.should.be.an('object');
+          newArticleAlert.createdArticle.should.have.property('slug');
+          newArticleAlert.createdArticle.should.have.property('title');
+          newArticleAlert.createdArticle.should.have.property('description');
```
@emekafredy I think you can make this code more consise by using destructuring to extract `createdArticle` from `newArticleAlert` as well. lines `114 - 121` and `139 - 142`

emekafredy replied:
Okay. I'm also working on this

lauragift21 commented on .hound.yml:
```diff
@@ -4,5 +4,5 @@ scss:
 eslint:
   enabled: true
-  config_file: .eslintrc
+  config_file: .eslintrc.json
```
@emekafredy You can revert the changes to this file. So it doesn't cause conflicts with what we have in the develop branch.

emekafredy replied:
Okay, thanks. I'm on it

emekafredy committed:
bug(PR): effect some changes on the branch

- Revert eslint change to avoid conflict
- ES6 object destructuring should be improved in the test file
- Correct some typographic errors

(Fixes #159987715)

emekafredy committed:
bug(tagModel): change 'name' attribute to 'tag' to avoid naming conflicts from other models
